app/src/main/java/com/kpiroom/bubble/ui/accountSetup/AccountSetupLogic.kt

```diff
@@ -0,0 +1,166 @@
+        clearUsernameFocus.value = true
+        photoChangeRequested.value = true
+        wallpaperChangeRequested.value = false
+    }
+
+    fun onWallpaperChanged() {
+        clearUsernameFocus.value = true
+        photoChangeRequested.value = false
+        wallpaperChangeRequested.value = true
+    }
+
+    fun onGetStarted() {
+        clearUsernameFocus.value = true
+        username.value.let {
+            when {
+                it.isNullOrBlank() -> {
+                    progress.alert(str(R.string.setup_username_cannot_be_empty))
+                    return
+                }
+                usernameExists(it) -> {
+                    progress.alert(str(R.string.setup_username_exists))
+                    return
+                }
+                else -> {
+                    if (it.length < 6) {
+                        progress.alert(str(R.string.setup_username_too_short))
+                        return
+                    } else if (it.length > 16) {
+                        progress.alert(str(R.string.setup_username_too_long))
+                        return
+                    } else if (!it.matches(usernameRegex)) {
+                        progress.alert(str(R.string.setup_username_invalid))
+                        return
+                    } else {
+                        if (photoDrw.value == null || wallpaperPhotoDrw.value == null) {
+                            progress.alert(
+                                getUnchangedDrwMessage(),
+                                ::finishSetupCondition
+                            )
+                        } else finishSetup()
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getUnchangedDrwMessage(): String? {
+        val photo = photoDrw.value
+        val wallpaper = wallpaperPhotoDrw.value
+
+        val unchangedDrwString = if (photo == null && wallpaper == null)
+            "photo and wallpaper"
+        else if (photo == null)
+            "photo"
+        else if (wallpaper == null)
+            "wallpaper"
+        else null
+
+        return app.getString(R.string.continue_with_unchanged, unchangedDrwString)
+    }
+
+    private fun finishSetupCondition(condition: Boolean) {
+        if (condition)
+            finishSetup()
+        else
+            return
+    }
+
+    private fun finishSetup() {
+        saveProfileData()
+        started.value = true
+    }
+
+    private fun saveProfileData() {
+        Source.userPrefs.let {
+            it.username = username.value
+            it.joinedDateString = dateString
+            it.profilePhotoUri = photoUri
+            it.profileWallpaperUri = wallpaperUri
+        }
+    }
+
+    private fun usernameExists(username: String): Boolean {
+        var exists = false
+        AsyncProcessor {
+            exists = Source.api.usernameExists(username)
+        } handleError {
+            progress.alert(it.message)
+        } runWith (bag)
+
+        return exists
```

Comment:
remove this. Show alert after asyncProcessor end up with usernameExists() result